### PR TITLE
Add testing for secured db + add and refactor configuration settings

### PR DESCRIPTION
**Pull request by lastav5**

### CODE REVIEW

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs [hunk 1]
```diff
@@ -1,86 +1,237 @@
...
+            DisposeStores();
         }
 
         public Config CreateSpecConfigAndStore(out IDocumentStore store)
         {
-            store = GetDocumentStore();
+            if (IsSecure)
+                store = GetDocumentStoreSecured(new X509Certificate2(CertificatePath), new());
+            else
+                store = GetDocumentStore(new());
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: what is that `new()` ?

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs [hunk 2]
```diff
@@ -1,86 +1,237 @@
...
+            else
+                _store = GetDocumentStore();
         }
-
+        
         public override void Dispose()
         {
             _store.Dispose();
             base.Dispose();
+            DisposeStores();
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: what about the `_store.Dispose();` above?

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs [hunk 3]
```diff
@@ -1,86 +1,237 @@
...
-                                        }
-                                    }
-                                }
-                                
-                                """;
+                                    snapshot-store {
+                                         plugin = "akka.persistence.snapshot-store.ravendb"
+                                         ravendb {
+                                             class = "Akka.Persistence.RavenDb.Snapshot.RavenDbSnapshotStore, Akka.Persistence.RavenDb"
+                                             auto-initialize = on
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: what is `auto-initialize` ?

File: src/Akka.Persistence.RavenDB/Journal/RavenDbJournal.cs [hunk 4]
```diff
@@ -1,22 +1,17 @@
...
-using Akka.Persistence.RavenDb.Query.ContinuousQuery;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
 
 namespace Akka.Persistence.RavenDb.Journal
 {
     public class RavenDbJournal : AsyncWriteJournal
     {
-        private readonly JournalRavenDbPersistence _storage = Context.System.WithExtension<JournalRavenDbPersistence, JournalRavenDbPersistenceProvider>();
+        private readonly JournalRavenDbPersistence _storage = Context.System.WithExtension<JournalRavenDbPersistence, JournalRavenDbPersistenceProvider>();//TODO what does this do
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB/Journal/RavenDbJournal.cs]: TODO?

File: src/Akka.Persistence.RavenDB/Journal/RavenDbJournal.cs [hunk 5]
```diff
@@ -99,6 +94,8 @@ private async Task AtomicWriteForActor(IGrouping<string, AtomicWrite> atomicWrit
             using var _ = await GetLocker(persistenceId).ReaderLockAsync(cts.Token);
             using var session = _storage.OpenAsyncSession();
             session.Advanced.SessionInfo.SetContext(persistenceId);
+            // session.Advanced.SetTransactionMode(mode);
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB/Journal/RavenDbJournal.cs:104]: remove this?

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs [hunk 6]
```diff
@@ -1,86 +1,237 @@
...
+            AsyncHelpers.RunSync(() => ImportDatabaseAsync(store, name));
+
+            SetupDatabase(store);
+
+            if (options?.WaitForIndexingTimeout.HasValue == true)
+                WaitForIndexing(store, name, options.WaitForIndexingTimeout);
+
+            _documentStores[store] = null;
+
+            return store;
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: instead all of this, would it make sense to override `PreInitialize` /`SetupDatabase` ?

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs [hunk 7]
```diff
@@ -1,86 +1,237 @@
...
+
+            async Task ImportDatabaseAsync(DocumentStore docStore, string database, TimeSpan? timeout = null)
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: why do we need that?

File: src/Akka.Persistence.RavenDB/Query/RavenDbReadJournal.cs [hunk 8]
```diff
@@ -1,20 +1,15 @@
...
-using Nito.AsyncEx;
-using Raven.Client.Documents.Linq;
-using Raven.Client.Documents.Subscriptions;
+using Akka.Streams.Dsl;
+using System.Threading.Channels;
 
 namespace Akka.Persistence.RavenDb.Query
 {
-    public class RavenDbReadJournal :
+    public class RavenDbReadJournal ://TODO why called Journal if namespace is Query
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB/Query/RavenDbReadJournal.cs:12]: it is `ReadJournal` and this the name in all other plugins, no?

File: src/Akka.Persistence.RavenDB/RavenDbConfiguration.cs [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+                {
+                    HttpVersion = Version.Parse(httpVersion);
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException(
+                        $"Could not parse 'http-version' configuration parameter value '{httpVersion}'.", e);
+                    //TODO stav: error doesn't bubble up when running test - only shows when debugging
+                }
+            }
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB/RavenDbConfiguration.cs]: why we wrap this in try/catch?

File: src/Akka.Persistence.RavenDB/RavenDbConfiguration.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        }
+
+        public DocumentConventions ToDocumentConventions()
+        {
+            var conventions = new DocumentConventions();
+
+            conventions.HttpVersion = HttpVersion ?? conventions.HttpVersion;
+            conventions.DisableTcpCompression = DisableTcpCompression ?? conventions.DisableTcpCompression;
+            conventions.RequestTimeout = RequestTimeout == TimeSpan.Zero ? conventions.RequestTimeout : RequestTimeout;
+            //conventions.Serialization =
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB/RavenDbConfiguration.cs]: ?

File: src/Akka.Persistence.RavenDB/RavenDbPersistence.cs [hunk 11]
```diff
@@ -26,61 +23,69 @@ public SnapshotRavenDbPersistence(ExtendedActorSystem system) : base(system)
...
         }
 
-        public static Config DefaultConfiguration()
-        {
-            return ConfigurationFactory.FromResource<RavenDbPersistence>("Akka.Persistence.RavenDb.reference.conf");
-        }
+        //public static Config DefaultConfiguration()
+        //{
+        //    return ConfigurationFactory.FromResource<RavenDbPersistence>("Akka.Persistence.RavenDb.reference.conf");
+        //}
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB/RavenDbPersistence.cs]: why this commented out?

File: src/Akka.Persistence.RavenDB/reference.conf [hunk 12]
```diff
@@ -13,27 +13,25 @@ akka.persistence {
...
-			# Default: 10 seconds
-			call-timeout = 10s
+			# Location of a client certificate to access a secure RavenDB database
+			#cert = "\\path\\to\\cert.pfx"
+
+			# Timeout for all requests sent to RavenDB.
+			# Client will fail requests that take longer than this.
+			request-timeout = 20s
+
+			#Http version for the RavenDB client to use in communication with the server
```
Comment (karmeli87) [inline src/Akka.Persistence.RavenDB/reference.conf:25]: let's say what the default here

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs (hunk 3 above)
Comment (lastav5) [reply src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: removed

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs (hunk 1 above)
Comment (lastav5) [reply src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: fixed

File: src/Akka.Persistence.RavenDB/Journal/RavenDbJournal.cs (hunk 4 above)
Comment (lastav5) [reply src/Akka.Persistence.RavenDB/Journal/RavenDbJournal.cs]: removed

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs (hunk 7 above)
Comment (lastav5) [reply src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: We probably don't. But it's something the original TestDriver does (private func) so I added here too. Gonna remove it for now

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs (hunk 6 above)
Comment (lastav5) [reply src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: Can't set the certificate inside these

File: src/Akka.Persistence.RavenDB/RavenDbConfiguration.cs (hunk 9 above)
Comment (lastav5) [reply src/Akka.Persistence.RavenDB/RavenDbConfiguration.cs]: removed

File: src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs (hunk 2 above)
Comment (lastav5) [reply src/Akka.Persistence.RavenDB.Tests/RavenDbFixture.cs]: no longer relevant

**Pull request merged**